File: lib/go/internal/types/TInteger.go
//|
//|  Copyright The Telepact Authors
//|  SPDX-License-Identifier: Apache-2.0
//|

package types

const integerName = "Integer"

// TInteger represents the Telepact integer type.
type TInteger struct{}

// NewTInteger constructs a TInteger instance.
func NewTInteger() *TInteger {
	return &TInteger{}
}

// GetTypeParameterCount implements TType.
func (t *TInteger) GetTypeParameterCount() int {
	return 0
}

// Validate implements TType.
func (t *TInteger) Validate(value any, typeParameters []*TTypeDeclaration, ctx *ValidateContext) []*ValidationFailure {
	return ValidateInteger(value)
}

// GenerateRandomValue implements TType.
func (t *TInteger) GenerateRandomValue(blueprintValue any, useBlueprintValue bool, typeParameters []*TTypeDeclaration, ctx *GenerateContext) any {
	return GenerateRandomInteger(blueprintValue, useBlueprintValue, ctx)
}

// GetName implements TType.
func (t *TInteger) GetName(ctx *ValidateContext) string {
	return integerName
}

File: lib/ts/src/internal/types/TInteger.ts
//|
//|  Copyright The Telepact Authors
//|  SPDX-License-Identifier: Apache-2.0
//|

import { ValidationFailure } from '../validation/ValidationFailure.js';
import { TTypeDeclaration } from './TTypeDeclaration.js';
import { TType } from './TType.js';
import { validateInteger } from '../validation/ValidateInteger.js';
import { generateRandomInteger } from '../generation/GenerateRandomInteger.js';
import { GenerateContext } from '../generation/GenerateContext.js';
import { ValidateContext } from '../validation/ValidateContext.js';

export const integerName: string = 'Integer';

export class TInteger extends TType {
    getTypeParameterCount(): number {
        return 0;
    }

    validate(value: any, typeParameters: TTypeDeclaration[], ctx: ValidateContext): ValidationFailure[] {
        return validateInteger(value);
    }

    generateRandomValue(
        blueprintValue: any,
        useBlueprintValue: boolean,
        typeParameters: TTypeDeclaration[],
        ctx: GenerateContext,
    ): any {
        return generateRandomInteger(blueprintValue, useBlueprintValue, ctx);
    }

    getName(): string {
        return integerName;
    }
}
